### PR TITLE
cleaning warnings from -Wall and -Wextra

### PR DESCRIPTION
**Pull request by sslattery**
Just a note that the following warning was fixed by commenting out the 2 variable names:
```
/Users/uy7/software/Cabana/core/src/Cabana_VerletList.hpp:163:27: warning: unused parameter 'begin' [-Wunused-parameter]
        const std::size_t begin,
                          ^
/Users/uy7/software/Cabana/core/src/Cabana_VerletList.hpp:164:27: warning: unused parameter 'end' [-Wunused-parameter]
        const std::size_t end,
                          ^
2 warnings generated.
```
this is directly related to #54.

Also adding stronger warnings and warnings as errors to the CI build.

### CODE REVIEW

**Comment by junghans** [comment]
We could add a `-Werror` build to the CI as well!

**Comment by sslattery** [comment]
I would be ok with `-Werror`. The last warning would need a temporary fix (i.e. commenting out the variable names on those two lines) so that the build would pass. I propose adding `-Wall -Wextra -Werror -pedantic` to the CI `CMAKE_CXX_FLAGS`.

File: .travis.yml [hunk 1]
```diff
@@ -43,6 +43,7 @@ addons:
       - libopenmpi-dev
 
 script:
+  - export CXXFLAGS="-Wall -Wextra -pedantic ${COVERAGE+:-Werror}"
```
[the remark this answers is not in the document]
Comment (junghans) [reply .travis.yml]: There is just no easy bash thing to do the opposite of `${COVERAGE+:-Werror}`, that is why I did it that way around. Something like `$([[ ${COVERAGE} ]] || echo -Werror)` would work though.

[the remark this answers is not in the document]
Comment (sslattery) [reply .travis.yml]: Can we give that a try then?

[the remark this answers is not in the document]
Comment (junghans) [reply .travis.yml]: Done! I introduced a `WERROR` variable to make it more clear.

**Comment by sslattery** [comment]
So apparently clang also fails with `-Werror` when there are build/link arguments that are not used; see the travis errors such as:

```

**Comment by junghans** [comment]
@sslattery here is the [magic](http://petereisentraut.blogspot.com/2011/09/ccache-and-clang-part-2.html) to fix it

**Pull request merged**